docs/reference/snapshot-restore/apis/get-snapshot-api.asciidoc

```diff
@@ -98,6 +98,36 @@ comprising the number of shards in the index, the total size of the index in
 bytes, and the maximum number of segments per shard in the index. Defaults to
 `false`, meaning that this information is omitted.
 
+`sort`::
+(Optional, string)
+Allows setting a sort order for the result. Defaults to `start_time`, i.e. sorting by snapshot start time stamp.
++
+.Valid values for `sort`
+[%collapsible%open]
+====
+`start_time`::
+  Sort snapshots by their start time stamp and break ties by snapshot name.
+
+`duration`::
+  Sort snapshots by their duration and break ties by snapshot name.
+
+`name`::
+  Sort snapshots by their name.
+
+`indices`::
+  Sort snapshots by the number of indices they contain and break ties by snapshot name.
+====
+
+`size`::
```

Comment:
Also, we talked about this before, we want an upper bound here probably in some implicit + explicit form. I think from a response size perspective I'd go with something like `200` for the upper limit on the size in case it's explicitly set and also return a `400` in case a request that does not have `size` set would return more than a certain limit number of snapshots. The only issue with this is that currently there are users for whom something like `3k` snapshots barely works and having an implicit limit would break their use case potentially.

Reply:
> Could we leave this untouched if none of sort, search, size are specified at least in 7.x?

I wonder if we can/should :) Obviously we don't want to break the API here, but we are moving towards creating situations where users will have a lot more snapshots that before. We did have problems with the memory consumption of the unpaginated API before and although it's better now, I think we are creating situations where a call getting all snapshot verbose could OOM/destabilize a master easily (thinking about shard failures in particular here). If we do keep the unlimited option untouched in 7.x (which makes sense don't get me wrong :)), should we then maybe invest some effort into some sort of circuit breaking during collecting the `SnapshotInfo` instances to prevent adding a widely used API that can OOM master easily (we could estimate memory from the amount of bytes we read from the blob store or so)?